manifests/validate_db_connection.pp

```diff
@@ -66,7 +66,7 @@
     default => "PGPASSWORD=${database_password_unsensitive}",
   }
   $cmd = join([$cmd_init, $cmd_host, $cmd_user, $cmd_port, $cmd_dbname], ' ')
-  $validate_cmd = "${validcon_script_path} ${sleep} ${tries} '${cmd}'"
+  $validate_cmd = [$validcon_script_path, $sleep, $tries, join(["'", $cmd, "'"], '')]
```

Comment:
Now you're basically doing 2 joins. This feels very complicated. Perhaps it's better to keep `$cmd` as an array and then deal with it here?

Reply:
Tried it in one but the problem is that they both need a different join value, i.e. ' ' and ''....
Suppose I could add the ' to the first and last variables, cmd_init and cmd_dbname but that just seems wrong to me

Reply:
Or do they... have been assuming but should double check that

Reply:
Ok yeh, looks like having a space between the ' and the values is fine, atleast according to the tests.
Not sure why I thought it wouldn't be